components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/constant/SQLConstants.java

```diff
@@ -72,6 +90,28 @@
             "(NAME), " +
             "LAST_MODIFIED = VALUES(LAST_MODIFIED), HAS_FILE = VALUES(HAS_FILE), HAS_ATTRIBUTE = VALUES" +
             "(HAS_ATTRIBUTE), TYPE_ID = VALUES(TYPE_ID)";
+    public static final String INSERT_OR_UPDATE_RESOURCE_POSTGRESQL = "INSERT INTO IDN_CONFIG_RESOURCE" +
+            "(ID,TENANT_ID,NAME, CREATED_TIME,LAST_MODIFIED,HAS_FILE, HAS_ATTRIBUTE,TYPE_ID) " +
+            "VALUES(?, ?, ?, ?, ?, ?, ?, ?) ON CONFLICT (ID) DO UPDATE " +
+            "SET TENANT_ID = EXCLUDED.TENANT_ID, NAME = EXCLUDED.NAME, LAST_MODIFIED = EXCLUDED.LAST_MODIFIED, " +
+            "HAS_FILE =EXCLUDED.HAS_FILE, HAS_ATTRIBUTE = EXCLUDED.HAS_ATTRIBUTE, TYPE_ID = EXCLUDED.TYPE_ID";
+    public static final String INSERT_OR_UPDATE_RESOURCE_DB2 = "MERGE INTO IDN_CONFIG_RESOURCE USING " +
+            "\"SYSIBM\".DUAL ON (ID= ? ) WHEN MATCHED THEN UPDATE SET TENANT_ID = ?, NAME = ?, LAST_MODIFIED = ?, " +
+            "HAS_FILE = ?, HAS_ATTRIBUTE = ?, TYPE_ID = ? WHEN NOT MATCHED THEN INSERT (ID,TENANT_ID,NAME, " +
+            "CREATED_TIME,LAST_MODIFIED,HAS_FILE, HAS_ATTRIBUTE,TYPE_ID) " +
+            "VALUES (?, ?, ?, ?, ?, ?, ?, ?)";
+    public static final String INSERT_OR_UPDATE_RESOURCE_MSSQL = "MERGE IDN_CONFIG_RESOURCE T USING " +
+            "(VALUES  (?, ?, ?, ?, ?, ?, ?, ?)) S " +
+            "(ID,TENANT_ID,NAME, CREATED_TIME,LAST_MODIFIED,HAS_FILE, HAS_ATTRIBUTE,TYPE_ID) ON T.ID = S.ID " +
+            "WHEN MATCHED THEN UPDATE SET TENANT_ID = S.TENANT_ID, NAME = S.NAME, LAST_MODIFIED = S.LAST_MODIFIED, " +
+            "HAS_FILE = S.HAS_FILE, HAS_ATTRIBUTE = S.HAS_ATTRIBUTE, TYPE_ID = S.TYPE_ID " +
+            "WHEN NOT MATCHED THEN INSERT (ID,TENANT_ID,NAME, CREATED_TIME,LAST_MODIFIED,HAS_FILE, HAS_ATTRIBUTE," +
+            "TYPE_ID) VALUES (S.ID, S.TENANT_ID, S.NAME, S.CREATED_TIME, S.LAST_MODIFIED, S.HAS_FILE, S" +
+            ".HAS_ATTRIBUTE, S.TYPE_ID);";
+    public static final String INSERT_OR_UPDATE_RESOURCE_ORACLE = "MERGE INTO IDN_CONFIG_RESOURCE USING dual ON " +
+            "(ID=?) WHEN MATCHED THEN UPDATE SET TENANT_ID=?, NAME=? , LAST_MODIFIED=?, HAS_FILE=?, HAS_ATTRIBUTE=?, " +
+            "TYPE_ID=? WHEN NOT MATCHED THEN INSERT (ID,TENANT_ID,NAME, CREATED_TIME, LAST_MODIFIED, HAS_FILE, " +
+            "HAS_ATTRIBUTE, TYPE_ID) VALUES (?,?, ?,?,?, ?, ?, ? )";
```

Comment:
formatting issue add space after ,

Reply:
addressed in 5a0504e41b864eb30d7aa555f60249f9b40e79e9